File: automator/src/main/java/com/stardust/automator/filter/StringStartsWithFilter.java
package com.stardust.automator.filter;

import com.stardust.automator.UiObject;

/**
 * Created by Stardust on 2017/3/9.
 */

public class StringStartsWithFilter extends DfsFilter {

    private final String mPrefix;
    private final KeyGetter mKeyGetter;

    public StringStartsWithFilter(String prefix, KeyGetter keyGetter) {
        mPrefix = prefix;
        mKeyGetter = keyGetter;
    }

    @Override
    protected boolean isIncluded(UiObject nodeInfo) {
        String key = mKeyGetter.getKey(nodeInfo);
        return key != null && key.startsWith(mPrefix);
    }

    @Override
    public String toString() {
        return mKeyGetter.toString() + "StartsWith(\"" + mPrefix + "\")";
    }
}

File: automator/src/main/java/com/stardust/automator/filter/StringStartsWithFilter.kt
package com.stardust.automator.filter

import com.stardust.automator.UiObject

/**
 * Created by Stardust on 2017/3/9.
 */

class StringStartsWithFilter(private val mPrefix: String, private val mKeyGetter: KeyGetter) : Filter {

    override fun filter(node: UiObject): Boolean {
        val key = mKeyGetter.getKey(node)
        return key != null && key.startsWith(mPrefix)
    }

    override fun toString(): String {
        return mKeyGetter.toString() + "StartsWith(\"" + mPrefix + "\")"
    }
}
